src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -66,22 +69,67 @@
         return getAcceptableTokens();
     }
 
+    /**
+     * Whether to allow default keyword not in last but surrounded with case.
+     * @param newValue whether to ignore checking.
+     */
+    public void setskipIfLastAndSharedWithCase(boolean newValue) {
+        skipIfLastAndSharedWithCase = newValue;
+    }
+
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
         //default keywords used in annotations too - not what we're
         //interested in
         if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
                 && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
-            final DetailAST switchAST = defaultGroupAST.getParent();
-            final DetailAST lastGroupAST =
-                switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            boolean containsInLastGroup = ast.getPreviousSibling() == null;
+            if (!skipIfLastAndSharedWithCase
+                    || containsInLastGroup) {
+                containsInLastGroup = containsInLastGroup(ast.getParent());
+            }
+            final boolean containsCase = containsCaseInBlock(ast);
+            if (containsCase || containsInLastGroup) {
                 log(ast, MSG_KEY);
```

Comment:
Look at discussion at https://github.com/checkstyle/checkstyle/pull/4084#discussion_r108250066
We can't use the same violation message. Current one makes no sense with new option. We need a new message for just this option.
New message with new option on should be: `Default should be last label in the case group.`
Old message needs to stay with new option off.